tap_universal_file/files.py

```diff
@@ -0,0 +1,199 @@
+            # Docs: https://filesystem-spec.readthedocs.io/en/latest/features.html#caching-files-locally
+            return fsspec.filesystem(
+                "filecache",
+                target_protocol=self.protocol,
+                target_options=self._get_args(),
+            )
+        if caching_strategy == "persistent":
+            return fsspec.filesystem(
+                "filecache",
+                target_protocol=self.protocol,
+                target_options=self._get_args(),
+                cache_storage=tempfile.gettempdir(),
+            )
+        if caching_strategy == "none":
+            # When caching is not used, the protocol's arguemnts have to be
+            # star-unpacked because fsspec.filesystem accepts them directly instead of
+            # as a dictionary.
+            return fsspec.filesystem(
+                protocol=self.protocol,
+                **self._get_args(),
+            )
+        msg = f"The caching strategy '{caching_strategy} is invalid."
+        raise ValueError(msg)
+
+    def get_files(
+        self,
+        starting_replication_key_value: str | None = None,
+    ) -> Generator[dict, None, None]:
+        """Gets file names to be synced.
+
+        Args:
+            starting_replication_key_value: _description_. Defaults to None.
+
+        Raises:
+            RuntimeError: If no files match the configured regex pattern or replication
+            key value.
+
+        Yields:
+            The name of a file to be synced, matching a regex pattern, if one has been
+                configured.
+        """
+        none_found = True
+        none_synced = True
+
+        file_dict_list = []
+
+        for file in self.filesystem.ls(self.config["filepath"], detail=True):
+            if (
+                file["type"] == "directory"  # Ignore nested folders.
+                or file["size"] == 0  # Ignore empty files.
+                or (  # Ignore files not matching the congigured file_regex
+                    "file_regex" in self.config
+                    and not re.match(
+                        self.config["file_regex"],
+                        Path(file["name"]).name,
+                    )
+                )
+            ):
+                continue
+            none_found = False
+            file_dict_list.append(
+                {"name": file["name"], "last_modified": self._get_last_modified(file)},
+            )
+
+        # Sort the files so that is_sorted can be True. This allows the tap to pick up
+        # where it left off if interrupted.
+        file_dict_list = sorted(file_dict_list, key=lambda k: k["last_modified"])
+
+        # Only yield files when no replication key is present or when the file is newer
+        # than the replication key value, as a datetime.
+        for file_dict in file_dict_list:
+            if starting_replication_key_value is None or file_dict[
+                "last_modified"
+            ] >= datetime.datetime.strptime(
```

Comment:
What do we think about making this a great than instead so we don't sync the same files twice? Any downsides here, I think with a file we can get away without having to worry about something updating at an identical time but maybe I"m mistaken here. Just seems odd to sync the same file when I uploaded them a month ago 

```suggestion
            ] > datetime.datetime.strptime(
```

Reply:
"Using the state, the tap should then skip returning rows where the replication key comes _strictly before_ than previous maximal replication key value stored in the state. [...] In incremental replication, it is OK and usually recommended to resend rows where the replication key is equal to previous highest key." from [Meltano documentation](https://sdk.meltano.com/en/latest/incremental_replication.html).

Reply:
hmmmm you have a valid point I think it's rare but we'll go with it for now